### PR TITLE
feat: #1408 improve icon picker design

### PR DESCRIPTION
**Pull request by manuel-rw**
<br/>
<div align="center">
  
  <h3>Homarr</h3>
</div>

**Thank you for your contribution. Please ensure that your pull request meets the following pull request:**

- [ ] Builds without warnings or errors (``pnpm buid``, autofix with ``pnpm format:fix``)
- [ ] Pull request targets ``dev`` branch
- [ ] Commits follow the [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] No shorthand variable names are used (eg. ``x``, ``y``, ``i`` or any abbrevation)

image

### CODE REVIEW

**Comment by Meierschlumpf** [comment]
When you plan to add the keyboard interactions?

File: apps/nextjs/src/components/icons/picker/icon-picker.tsx [hunk 1]
```diff
@@ -35,33 +35,48 @@ export const IconPicker = ({ initialValue, onChange, error, onFocus, onBlur }: I
 
   const groups = notNullableData.map((group) => {
```
Comment (Meierschlumpf) [inline apps/nextjs/src/components/icons/picker/icon-picker.tsx:36]: You could try if the `useSuspenseQuery` also works for this one, then we wouldn't need to fallback to an empty array

File: apps/nextjs/src/components/icons/picker/icon-picker.tsx [hunk 2]
```diff
@@ -35,33 +35,48 @@ export const IconPicker = ({ initialValue, onChange, error, onFocus, onBlur }: I
...
-      </Combobox.Option>
+      <Card
+        onClick={() => {
+          const value = item.url;
+          setValue(value);
+          setPreviewUrl(value);
+          setSearch(value);
+          onChange(value);
+          combobox.closeDropdown();
+        }}
```
Comment (Meierschlumpf) [inline apps/nextjs/src/components/icons/picker/icon-picker.tsx:46]: Please wrap this with an unstyled button!

File: apps/nextjs/src/components/icons/picker/icon-picker.tsx [hunk 3]
```diff
@@ -35,33 +35,48 @@ export const IconPicker = ({ initialValue, onChange, error, onFocus, onBlur }: I
...
-          <Text>{item.name}</Text>
-        </Group>
-      </Combobox.Option>
+      <Card
+        onClick={() => {
+          const value = item.url;
+          setValue(value);
+          setPreviewUrl(value);
+          setSearch(value);
+          onChange(value);
```
Comment (Meierschlumpf) [inline apps/nextjs/src/components/icons/picker/icon-picker.tsx:44]: previewUrl and value seem to have the same values.

File: apps/nextjs/src/components/icons/picker/icon-picker.tsx [hunk 4]
```diff
@@ -35,33 +35,48 @@ export const IconPicker = ({ initialValue, onChange, error, onFocus, onBlur }: I
...
+          combobox.closeDropdown();
```
Comment (Meierschlumpf) [inline apps/nextjs/src/components/icons/picker/icon-picker.tsx:45]: You can wrap those in a 
```ts
startTransition(() => {
   // Place your 4 setters here
});
```

https://react.dev/reference/react/startTransition

File: apps/nextjs/src/components/icons/picker/icon-picker.tsx [hunk 5]
```diff
@@ -35,33 +35,48 @@ export const IconPicker = ({ initialValue, onChange, error, onFocus, onBlur }: I
...
+        }}
+      >
+        <Box w={25} h={25}>
+          <Image src={item.url} w={25} h={25} radius="md" />
+        </Box>
+        {item.url.endsWith(".svg") && (
+          <Badge pos="absolute" top={0} right={0} style={{ transform: "translate3d(5px, -5px, 0)" }} size="sm">
+            SVG
+          </Badge>
+        )}
```
Comment (Meierschlumpf) [inline apps/nextjs/src/components/icons/picker/icon-picker.tsx:62]: I still prefer the usage of a indicator that already has it's placing implemented and can be disabled with the disabled prop

File: apps/nextjs/src/components/icons/picker/icon-picker.tsx [hunk 6]
```diff
@@ -35,33 +35,48 @@ export const IconPicker = ({ initialValue, onChange, error, onFocus, onBlur }: I
...
 
     return (
-      <Combobox.Group label={group.slug} key={group.id}>
-        {options}
-      </Combobox.Group>
+      <Paper p="xs" key={group.slug} pt={2}>
+        <Text mb={6} size="sm" fw="bold">
+          {group.slug}
+        </Text>
+        <Box display="flex" style={{ gap: 8, flexWrap: "wrap" }}>
```
Comment (Meierschlumpf) [inline apps/nextjs/src/components/icons/picker/icon-picker.tsx:71]: Maybe instead use `Flex` component

File: packages/validation/src/icons.ts [hunk 7]
```diff
@@ -2,6 +2,7 @@ import { z } from "zod";
 
 const findIconsSchema = z.object({
   searchText: z.string().optional(),
+  limitPerGroup: z.number().min(0).max(500).default(12),
```
Comment (Meierschlumpf) [inline packages/validation/src/icons.ts:5]: Change min to 1

**Review by Meierschlumpf — CHANGES_REQUESTED** [review]

File: apps/nextjs/src/components/icons/picker/icon-picker.tsx (hunk 3 above)
Comment (manuel-rw) [reply apps/nextjs/src/components/icons/picker/icon-picker.tsx:44]: That is not true. See ``onChange``

File: apps/nextjs/src/components/icons/picker/icon-picker.tsx [hunk 8]
```diff
@@ -18,56 +32,57 @@ export const IconPicker = ({ initialValue, onChange, error, onFocus, onBlur }: I
...
-        <Box w={50} h={50}>
-          <Image src={item.url} w={50} h={50} radius="md" />
-        </Box>
-        {item.url.endsWith(".svg") && (
-          <Badge pos="absolute" top={0} right={0} style={{ transform: "translate3d(5px, -8px, 0)" }} size="sm">
-            SVG
-          </Badge>
-        )}
-      </Card>
+        <Indicator label="SVG" disabled={item.url.endsWith(".svg")} size={16}>
```
Comment (Meierschlumpf) [inline apps/nextjs/src/components/icons/picker/icon-picker.tsx:60]: Should be `disabled={!item.url.endsWith(".svg")}`

File: apps/nextjs/public/images/apps/imdb.png
Comment (Meierschlumpf) [inline apps/nextjs/public/images/apps/imdb.png:1]: This image seems to be broken as well (I would guess LF)

**Review by Meierschlumpf — CHANGES_REQUESTED** [review]

**Review by Meierschlumpf — APPROVED** [review]

**Pull request merged**